### PR TITLE
fix - kill network capture process if any container is killed

### PR DESCRIPTION
**Pull request by spiderxm**
## Description

Fix network thread issue

Issue is network thread is not terminating after an error
to reproduce the issue, run a module, and then kill the container with docker kill container_name. OHP is expected to exit from working and throw an error, but the network thread stays open and it's not terminating.

## Steps to Reproduce

```sh
# run the command below
$ sudo python3 ohp.py -m ftp/weak_password,ftp/strong_password --store-pcap
# wait until network capture process starts
```
```sh
# on new terminal run the command below
$ docker kill ohp_ftpserver_strong_password
# on the terminal where you started the capture process it will not end but it should.
```

## Related Issue

#359 

## Screenshots (if appropriate):
Screenshot 2022-08-02 at 10 18 45 AM
After killing the container the network capture process terminates successfully.
Screenshot 2022-08-02 at 10 18 52 AM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

#### Checklist
- [x] I have followed the [Contributor Guidelines](https://github.com/OWASP/Nettacker/wiki/Developers#contribution-guidelines).
- [x] The code has been thoroughly tested in my local development environment with flake8 and pylint.
- [x] The code is Python 3 compatible.
- [x] The code follows the PEP8 styling guidelines with 4 spaces indentation.
- [x] This Pull Request relates to only one issue or only one feature
- [x] I have referenced the corresponding issue number in my commit message
- [x] I have added the relevant documentation.
- [x] My branch is up-to-date with the Upstream master branch.

### CODE REVIEW

File: core/load.py [hunk 1]
```diff
@@ -362,13 +362,16 @@ def wait_until_interrupt(virtual_machine_container_reset_factory_time_seconds, c
                 # start containers based on selected modules
                 start_containers(configuration)
             if not new_network_events_thread.is_alive():
-                return error(messages["interrupt_application"])
+                error(messages["interrupt_application"])
+                return True
```
Comment (dhirensr) [inline core/load.py:366]: return in a while loop?

File: core/load.py [hunk 2]
```diff
@@ -884,6 +887,9 @@ def load_honeypot_engine():
     )
     # killed the network traffic capture process by ctrl + c... waiting to end.
     info(messages["killing_capture_process"])
+    if exit_flag:
+        # Terminate the network capture process
+        network_traffic_capture_process.terminate()
```
Comment (dhirensr) [inline core/load.py:892]: this should do the job right without the above changes?

File: core/load.py (hunk 1 above)
Comment (spiderxm) [reply core/load.py:366]: Return error("") 
Is returning None

So we need to return True in failing cases

We can break the while loop and at the end return True
What say @dhirensr ?

Comment (dhirensr) [reply core/load.py:366]: we can't use break there. sounds like  a use case for break?

File: core/load.py (hunk 2 above)
Comment (spiderxm) [reply core/load.py:892]: nopes, we need a exit flag value which denotes this has exited

**Comment by spiderxm** [comment]
@dhirensr check now

**Review by dhirensr — APPROVED** [review]

**Pull request merged**